dpcomp_gpu_runtime/lib/gpu_runtime.cpp

```diff
@@ -346,9 +346,7 @@ struct Stream {
       throw std::runtime_error("Failed to allocate MemInfo");
     }
 
-#if defined(IMEX_ENABLE_NUMBA_HOTFIX)
     retain();
```

Comment:
this was throwing an exception IIRC, I think this is used for reference counting for your python stuff and we dont require it

Reply:
It should only cause memory leak and not crash, GPU tests pass locally for me, can you check it on your side?

Reply:
yes, it gives a memory leak